app/controllers/application_controller.rb

```diff
@@ -1,3 +1,10 @@
 class ApplicationController < ActionController::Base
-  protect_from_forgery with: :exception
+ protect_from_forgery with: :exception
+ before_action :create_cart
```

Comment:
This is a solution we ended up using in the PhotoMarket application but I would be very careful in reusing it here.